### PR TITLE
WI #2193 Integrate Commit Statement

### LINKED ISSUE #2193: SQL COMMIT
**Decription**

Complete the [COMMIT](https://www.ibm.com/docs/en/db2-for-zos/12?topic=statements-commit) Syntax .

This implies:

- Ensure that all Keywords are recognized by the sql scanner
- Store all tokens in the SqlObject instance
- Update unit tests

### PR DESCRIPTION
**Pull request by laabidiHend**
Fixes #2193

### CODE REVIEW

File: TypeCobol.Test/Parser/Programs/Cobol85/ExecSql/ExecSqlWithCommit.rdz-Tokens.txt
Comment (fm-117) [inline TypeCobol.Test/Parser/Programs/Cobol85/ExecSql/ExecSqlWithCommit.rdz-Tokens.txt:1]: That should be ok after merging #2189, right ?

File: TypeCobol.Test/Parser/Programs/Cobol85/ExecSql/ExecSqlWithCommit.rdz.cbl
Comment (fm-117) [inline TypeCobol.Test/Parser/Programs/Cobol85/ExecSql/ExecSqlWithCommit.rdz.cbl:9]: No. We need to keep the original test line. The idea is to test unusual cases like this one where two EXEC SQL are on the same line.

Just add your new test cases after the existing one, so at line 11 you can add your new `EXEC SQL COMMIT WORK`.

File: TypeCobol.Test/Utils/Comparator.cs
Comment (fm-117) [inline TypeCobol.Test/Utils/Comparator.cs:208]: No need to comment code, remove it. Commented code brings confusion for readers, we'll alaways have git history to see previous modifications

File: TypeCobol/AntlrGrammar/CobolCodeElements.g4
Comment (fm-117) [inline TypeCobol/AntlrGrammar/CobolCodeElements.g4:8263]: Yes, it can even be improved a bit. see my remark in #2209 about ddefining it as a re-usable contextual keyword.

**Review by fm-117 — CHANGES_REQUESTED** [review]

File: TypeCobol.Test/Parser/Programs/Cobol85/ExecSql/ExecSqlWithCommit.rdz-Tokens.txt
Comment (laabidiHend) [reply TypeCobol.Test/Parser/Programs/Cobol85/ExecSql/ExecSqlWithCommit.rdz-Tokens.txt:1]: yes

**Review by fm-117 — APPROVED** [review]
This PR will be impacted by #2189 and #2209, so better keep it open right now and wait for the merge of the other two.

File: TypeCobol/AntlrGrammar/CobolCodeElements.g4
Comment (mayanje) [inline TypeCobol/AntlrGrammar/CobolCodeElements.g4:8265]: @laabidiHend Can you please add a test with a COMMIT statement wich is followed by another identifier than WORK. Just to check that a syntax error is reported.

**Review by smedilol — APPROVED** [review]

**Comment by fm-117** [comment]
Re-requesting review as we had to introduce a new method to help ANTLR recover from parsing errors while in SQL context.

File: TypeCobol/Compiler/AntlrUtils/CobolErrorStrategy.cs
Comment (smedilol) [inline TypeCobol/Compiler/AntlrUtils/CobolErrorStrategy.cs:298]: We can also delete `nextToken.TokenFamily == TokenFamily.CompilerDirectiveStartingKeyword` I think.

File: TypeCobol/Compiler/AntlrUtils/CobolErrorStrategy.cs
Comment (smedilol) [inline TypeCobol/Compiler/AntlrUtils/CobolErrorStrategy.cs:296]: Do we need a special for `SQL_FROM` ? 
The test `SqlStatementElement.IsSqlStatementStartingKeyword(nextToken.TokenType)` should be enough, no ?

**Review by smedilol — CHANGES_REQUESTED** [review]

**Review by mayanje — APPROVED** [review]

File: TypeCobol/Compiler/AntlrUtils/CobolErrorStrategy.cs
Comment (fm-117) [reply TypeCobol/Compiler/AntlrUtils/CobolErrorStrategy.cs]: Definitely ! Forgot that after writing it as a test.

File: TypeCobol/Compiler/AntlrUtils/CobolErrorStrategy.cs
Comment (fm-117) [reply TypeCobol/Compiler/AntlrUtils/CobolErrorStrategy.cs]: Definitely ! Forgot that after writing it as a test.

**Pull request merged**
